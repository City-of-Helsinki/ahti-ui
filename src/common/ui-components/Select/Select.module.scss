@import '../../../styles/spacings';

$white: var(--hds-ui-color-white);

.select {
  display: inline-block;
  font-size: 18px;
  line-height: 1.15;
  padding: unit(1) unit(0.75);
  width: 100%;
  max-width: 100%;
  box-sizing: border-box;
  margin: 0;
  border: 2px solid var(--hds-ui-color-black-50);
  border-radius: 0;
  background-color: var(--hds-ui-color-white);
  background-image: url('./angleDown.svg'), linear-gradient($white, $white);
  background-repeat: no-repeat, repeat;
  background-position: right unit(1) top 50%, 0 0;
  background-size: 1em auto, 100%;
  appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;

  &::-ms-expand {
    display: none;
  }

  &:focus {
    border-color: var(--hds-ui-color-black);
  }

  &.disabled {
    background: var(--hds-ui-color-black-10);
    border-color: var(--hds-ui-color-black-10);
    color: var(--hds-ui-color-black-70);
  }
}

.select option {
  font-weight: normal;
}

.labelText {
  color: var(--label-color-default);
  display: block;
  font-size: var(--fontsize-subtitle);
  font-weight: 700;
  margin-bottom: var(--spacing-3-xs);
}
