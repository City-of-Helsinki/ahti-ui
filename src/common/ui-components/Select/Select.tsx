import React from 'react';
import classNames from 'classnames';

import styles from './Select.module.scss';

interface Option {
  label: string;
  value: string | number;
}

export type SelectProps = {
  name?: string;
  labelText?: string;
  className?: string;
  value: Option['value'] | undefined;
  options: Option[];
  required?: boolean;
  disabled?: boolean;
  onChange: React.ChangeEventHandler<HTMLSelectElement>;
  id?: string;
};

const Select: React.FC<SelectProps> = ({
  name,
  id,
  labelText,
  className,
  value,
  options,
  onChange,
  required,
  disabled,
}) => {
  const optionsItems = options.map(({ value, label }) => (
    <option key={value} value={value}>
      {label}
    </option>
  ));

  return (
    <div className={className}>
      {labelText && (
        <label className={styles.labelText} htmlFor={id}>
          {labelText}
        </label>
      )}
      <select
        name={name}
        id={id}
        value={value}
        onChange={onChange}
        disabled={disabled}
        className={classNames(styles.select, { [styles.disabled]: disabled })}
      >
        {!required && <option>-</option>}
        {optionsItems}
      </select>
    </div>
  );
};

export default Select;
