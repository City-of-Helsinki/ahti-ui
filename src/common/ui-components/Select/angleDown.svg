<?xml version="1.0" encoding="utf-8"?>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 199 199">
<g>
<polygon points="199.404,63.993 171.12,35.709 99.702,107.127 28.284,35.709 0,63.993 99.702,163.695  " />
</g>
</svg>