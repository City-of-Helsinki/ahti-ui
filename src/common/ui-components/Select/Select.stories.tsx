import React from 'react';
import { action } from '@storybook/addon-actions';

import Select from './Select';

export default {
  component: Select,
  title: 'Select',
};

export const select = () => (
  <Select
    labelText="Foo"
    value={undefined}
    options={[
      { value: 'foo', label: 'foo' },
      { value: 'bar', label: 'bar' },
    ]}
    onChange={action('changed')}
  />
);

select.story = {
  name: 'Default',
};
