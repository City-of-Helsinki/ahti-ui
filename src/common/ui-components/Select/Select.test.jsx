import React from 'react';
import { shallow } from 'enzyme';

import Select from './Select';

const mockProps = {
  labelText: 'Selector',
  value: 'first',
  options: [
    { value: 1, label: 'First' },
    { value: 2, label: 'Second' },
  ],
  onChange: jest.fn(),
};

describe('Select', () => {
  beforeEach(() => {
    jest.restoreAllMocks();
  });

  const getWrapper = (props = {}) => shallow(<Select {...mockProps} {...props} />);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
