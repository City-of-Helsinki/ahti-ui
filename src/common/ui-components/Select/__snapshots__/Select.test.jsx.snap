// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders normally 1`] = `"<label><span class=\\"labelText\\">Selector</span><select class=\\"select\\"><option>-</option><option value=\\"1\\">First</option><option value=\\"2\\">Second</option></select></label>"`;
